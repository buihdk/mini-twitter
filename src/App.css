.alert {
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  border-radius: unset;
  z-index: 999;
}

.shown {
  opacity: 0.85;
  transition: all 250ms linear;
}

.hidden {
  opacity: 0;
  transition: all 250ms linear;
}

header {
  background-image: url("background.jpg");
  background-position: center center;
  background-size: cover;
  height: 450px;
  color: white;
}

.avatar {
  width: 200px;
  border: 6px white solid;
  border-radius: 50%;
  position: relative;
  top: -6rem;
}

.tweetbox {
  height: 60px;
  padding: 10px;
  width: 100%;
  background-color: #fff;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
}

.tweet {
  background-color: white;
  padding: 1rem;
  box-shadow: 0 2px 3px rgba(10, 10, 10, 0.1), 0 0 0 1px rgba(10, 10, 10, 0.1);
  word-wrap: break-word;
}

.avatar-tweet {
  width: 64px;
  border-radius: 50%;
}

.fa-retweet,
.fa-heart,
.fa-trash-alt,
.sub-text {
  color: lightgray;
  font-size: 16px;
}

.fa-retweet:hover,
.fa-heart:hover,
.fa-trash-alt:hover {
  color: gray;
  cursor: pointer;
}

.fa-heart.liked {
  color: red;
}

@media (max-width: 576px) { 
  header {
    height: 300px;
  }
  .mobile-hidden {
    display: none;
  }
}