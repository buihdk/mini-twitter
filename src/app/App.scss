header {
  background-image: url("./background.jpg");
  background-position: center center;
  background-size: cover;
  height: 450px;
  color: white;
}

@media (max-width: 576px) { 
  header {
    height: 300px;
  }

  .mobile-hidden {
    display: none;
  }
}