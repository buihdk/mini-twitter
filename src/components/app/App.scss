.alert {
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  border-radius: unset;
  z-index: 999;
}

.shown {
  opacity: 0.85;
  transition: all 250ms linear;
}

.hidden {
  opacity: 0;
  transition: all 250ms linear;
}

header {
  background-image: url("./background.jpg");
  background-position: center center;
  background-size: cover;
  height: 450px;
  color: white;
}

@media (max-width: 576px) { 
  header {
    height: 300px;
  }

  .mobile-hidden {
    display: none;
  }
}