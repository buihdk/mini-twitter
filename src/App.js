import React, { Component } from 'react';
import './App.css';
import TweetBox from './TweetBox';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tweetAlert: false,
      deleteAlert: false,
      retweetAlert: false
    };
  }
  handleTweetAlert() {
    this.setState({ tweetAlert: true });
    setTimeout(() => { this.setState({ tweetAlert: false }); }, 2000);
  }
  handleDeleteAlert() {
    this.setState({ deleteAlert: true });
    setTimeout(() => { this.setState({ deleteAlert: false }); }, 2000);
  }
  handleRetweetAlert() {
    this.setState({ retweetAlert: true });
    setTimeout(() => { this.setState({ retweetAlert: false }); }, 2000);
  }

  render() {
    return (
      <div className="App">
        <div className={`alert alert-success ${this.state.tweetAlert ? 'shown' : 'hidden'}`}>
          Your Tweet was sent!
        </div>
        <div className={`alert alert-danger ${this.state.deleteAlert ? 'shown' : 'hidden'}`}>
          Your Tweet has been deleted!
        </div>
        <div className={`alert alert-info ${this.state.retweetAlert ? 'shown' : 'hidden'}`}>
          Your Tweet has been retweeted!
        </div>
        <header/>
        <TweetBox
          placeholder="What's your status?" 
          handleTweetAlert={this.handleTweetAlert.bind(this)}
          handleDeleteAlert={this.handleDeleteAlert.bind(this)}
          handleRetweetAlert={this.handleRetweetAlert.bind(this)}
        />
      </div>
    );
  }
}

export default App;
