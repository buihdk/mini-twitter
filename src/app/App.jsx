import React, { useState, memo } from 'react';

import TweetBox from '../components/tweet/TweetBox';
import CustomAlert from '../components/alert/CustomAlert';
import Snow from '../components/snow/Snow';
import Rain from '../components/rain/Rain';
import * as alertProps from '../constants';
import './App.scss';

const App = () => {
  const [state, setState] = useState({ isSnow: true, activeAlert: '' });

  const handleAlert = type => {
    setState(prevState => ({ ...prevState, activeAlert: type }));
    setTimeout(() => {
      setState(prevState => ({ ...prevState, activeAlert: '' }));
    }, 2000);
  };

  const { isSnow, activeAlert } = state;

  return (
    <div className="App">
      {isSnow ? <Snow /> : <Rain />}
      <header />
      <TweetBox placeholder="What's your status?" handleAlert={handleAlert} />
      <CustomAlert isShown={activeAlert} {...alertProps[activeAlert]} />
    </div>
  );
};

export default memo(App);
