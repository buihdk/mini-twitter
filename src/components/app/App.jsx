import React, { useState, memo } from 'react';
import Alert from 'react-bootstrap/Alert';

import TweetBox from '../tweet/TweetBox';
import Snow from '../snow/Snow';
import Rain from '../rain/Rain';
import './App.scss';

const App = () => {
  const [state, setState] = useState({
    weatherSnowOrRain: true,
    alert: {
      isTweet: false,
      isRetweet: false,
      isDelete: false,
    },
  });

  const handleAlert = type => {
    setState(prevState => {
      return { ...prevState, alert: { ...prevState.alert, [type]: true } };
    });
    setTimeout(() => {
      setState(prevState => {
        return { ...prevState, alert: { ...prevState.alert, [type]: false } };
      });
    }, 2000);
  };

  const {
    alert: { isTweet, isDelete, isReTweet },
    weatherSnowOrRain,
  } = state;
  return (
    <div className="App">
      <Alert className={isTweet ? 'shown' : 'hidden'} variant="success">
        Your tweet was sent.
      </Alert>
      <Alert className={isDelete ? 'shown' : 'hidden'} variant="danger">
        Your tweet has been deleted.
      </Alert>
      <Alert className={isReTweet ? 'shown' : 'hidden'} variant="info">
        Your tweet has been retweeted.
      </Alert>
      {weatherSnowOrRain ? <Snow /> : <Rain />}
      <header />
      <TweetBox placeholder="What's your status?" handleAlert={handleAlert} />
    </div>
  );
};

export default memo(App);
