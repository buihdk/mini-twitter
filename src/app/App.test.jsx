import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { act } from 'react-dom/test-utils';

import App from './App';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('App', () => {
  let wrapper;
  act(() => {
    wrapper = mount(<App />, container);
  });

  test('renders snow animation without crashing', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('renders rain animation without crashing', () => {
    wrapper.find('#toggle').simulate('click');

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
