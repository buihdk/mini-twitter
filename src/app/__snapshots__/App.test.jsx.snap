// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="snow"
  >
    <svg
      preserveAspectRatio="xMidYMax slice"
      viewBox="0 0 1024 1536"
    >
      <g
        fill="#FFF"
        fillOpacity=".15"
        transform="translate(55 42)"
      >
        <g
          id="snow-bottom-layer"
        >
          <ellipse
            cx="6"
            cy="1009.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="138"
            cy="1110.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="398"
            cy="1055.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="719"
            cy="1284.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="760"
            cy="1155.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="635"
            cy="1459.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="478"
            cy="1335.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="322"
            cy="1414.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="247"
            cy="1234.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="154"
            cy="1425.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="731"
            cy="773.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="599"
            cy="874.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="339"
            cy="819.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="239"
            cy="1004.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="113"
            cy="863.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="102"
            cy="1223.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="395"
            cy="1155.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="826"
            cy="943.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="626"
            cy="1054.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="887"
            cy="1366.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="6"
            cy="241.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="138"
            cy="342.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="398"
            cy="287.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="719"
            cy="516.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="760"
            cy="387.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="635"
            cy="691.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="478"
            cy="567.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="322"
            cy="646.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="247"
            cy="466.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="154"
            cy="657.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="731"
            cy="5.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="599"
            cy="106.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="339"
            cy="51.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="239"
            cy="236.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="113"
            cy="95.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="102"
            cy="455.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="395"
            cy="387.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="826"
            cy="175.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="626"
            cy="286.5"
            rx="6"
            ry="5.5"
          />
          <ellipse
            cx="887"
            cy="598.5"
            rx="6"
            ry="5.5"
          />
        </g>
      </g>
      <g
        fill="#FFF"
        fillOpacity=".3"
        transform="translate(65 63)"
      >
        <g
          id="snow-top-layer"
        >
          <circle
            cx="8"
            cy="776"
            r="8"
          />
          <circle
            cx="189"
            cy="925"
            r="8"
          />
          <circle
            cx="548"
            cy="844"
            r="8"
          />
          <circle
            cx="685"
            cy="1115"
            r="8"
          />
          <circle
            cx="858"
            cy="909"
            r="8"
          />
          <circle
            cx="874"
            cy="1438"
            r="8"
            transform="rotate(180 874 1438)"
          />
          <circle
            cx="657"
            cy="1256"
            r="8"
            transform="rotate(180 657 1256)"
          />
          <circle
            cx="443"
            cy="1372"
            r="8"
            transform="rotate(180 443 1372)"
          />
          <circle
            cx="339"
            cy="1107"
            r="8"
            transform="rotate(180 339 1107)"
          />
          <circle
            cx="24"
            cy="1305"
            r="8"
            transform="rotate(180 24 1305)"
          />
          <circle
            cx="8"
            cy="8"
            r="8"
          />
          <circle
            cx="189"
            cy="157"
            r="8"
          />
          <circle
            cx="548"
            cy="76"
            r="8"
          />
          <circle
            cx="685"
            cy="347"
            r="8"
          />
          <circle
            cx="858"
            cy="141"
            r="8"
          />
          <circle
            cx="874"
            cy="670"
            r="8"
            transform="rotate(180 874 670)"
          />
          <circle
            cx="657"
            cy="488"
            r="8"
            transform="rotate(180 657 488)"
          />
          <circle
            cx="443"
            cy="604"
            r="8"
            transform="rotate(180 443 604)"
          />
          <circle
            cx="339"
            cy="339"
            r="8"
            transform="rotate(180 339 339)"
          />
          <circle
            cx="24"
            cy="537"
            r="8"
            transform="rotate(180 24 537)"
          />
        </g>
      </g>
    </svg>
  </div>
  <header />
  <section
    className="tweetbox border-bottom"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="d-none d-md-block col-md-3"
        >
          <img
            className="avatar rounded-circle"
            src="https://pbs.twimg.com/profile_images/553467511211970560/nBE77dF0_400x400.jpeg"
          />
        </div>
        <div
          className="col-md-9 col-sm-12"
        >
          <div
            className="px-4 input-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="What's happening?"
              value=""
            />
            <div
              className="input-group-append"
            >
              <button
                className="btn btn-primary"
                disabled={false}
                onClick={[Function]}
                type="submit"
                value="submit"
              >
                <span
                  className="mobile-hidden"
                >
                  Tweet 
                </span>
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-twitter fa-w-16 fa-null fa-rotate-null fa-pull-null "
                  data-icon="twitter"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
              <p
                className="chars my-auto mobile-hidden"
              >
                140 chars left
              </p>
            </div>
          </div>
          <div
            className="tweets mt-4 container"
          />
        </div>
      </div>
    </div>
  </section>
  <div
    className="fade custom-alert hidden alert show"
    role="alert"
  >
    
  </div>
</div>
`;
